core/lib/dal/src/blocks_web3_dal.rs

```diff
@@ -568,10 +569,11 @@ impl BlocksWeb3Dal<'_, '_> {
 
         // FIXME (PLA-728): remove after 2nd phase of `fee_account_address` migration
         #[allow(deprecated)]
-        self.storage
-            .blocks_dal()
-            .maybe_load_fee_address(&mut details.operator_address, details.number)
-            .await?;
+        BlocksDal {
+            storage: self.storage,
+        }
```

Comment:
Dumb question: Could `Dal` structs wrap `&mut StorageProcessor` instead? (Obviously, it would make sense to implement `fn conn(&mut self) -> &mut PgConnection` to minimize the changeset, and `DerefMut<Target = BasicStorageProcessor` for the same purpose; although in the latter case, I'd argue that replacing `self.storage` with `&mut self.storage.0` doesn't make the PR *that* harder to review.) Having to construct DALs manually looks quite clunky.

Reply:
It can't wrap StorageProcessor - that's the most problematic part of this implementation. Since we don't want to actually separate types of `ConnectionPool -> StorageProcessor` in this PR(that is why this exact PR was made) - we aren't able to implement these `get_dal` methods for StorageProcessor because we will have some cyclic dependencies there - so we are using one type that we have inside of Dal structs - and the other one to actually be able to access Dal structs. I agree that it doesn't look good though - but I couldn't find a better way to implement this.